qt/src/hocr/HOCRPdfExporter.cc

```diff
@@ -684,3 +712,43 @@ QImage HOCRPdfExporter::getSelection(const QRect& bbox)
 {
 	return m_displayerTool->getSelection(bbox);
 }
+
+void HOCRPdfExporter::paperSizeChanged() {
+	delete ui.lineEditPaperSizeHeight->validator();
+	delete ui.lineEditPaperSizeWidth->validator();
+
+	if(ui.comboBoxPaperSizeMetric->itemText(ui.comboBoxPaperSizeMetric->currentIndex()) == "mm") {
+		ui.lineEditPaperSizeHeight->setValidator(new QIntValidator(0, 10000000));
+		ui.lineEditPaperSizeWidth->setValidator(new QIntValidator(0, 10000000));
+	} else if(ui.comboBoxPaperSizeMetric->itemText(ui.comboBoxPaperSizeMetric->currentIndex()) == "inch") {
+		ui.lineEditPaperSizeHeight->setValidator(new QDoubleValidator(0.0, 10000000.0, 2));
+		ui.lineEditPaperSizeWidth->setValidator(new QDoubleValidator(0.0, 10000000.0, 2));
+	}
+
+	auto textPaperSize = ui.comboBoxPaperSize->itemText(ui.comboBoxPaperSize->currentIndex());
+	if(textPaperSize == "Custom") {
+		ui.lineEditPaperSizeWidth->setEnabled(true);
+		ui.lineEditPaperSizeHeight->setEnabled(true);
+	} else if(textPaperSize == "Same as source") {
+		ui.lineEditPaperSizeWidth->setText("");
+		ui.lineEditPaperSizeHeight->setText("");
+		ui.lineEditPaperSizeWidth->setEnabled(false);
+		ui.lineEditPaperSizeHeight->setEnabled(false);
+	} else {
+		ui.lineEditPaperSizeWidth->setEnabled(false);
+		ui.lineEditPaperSizeHeight->setEnabled(false);
+		if(ui.comboBoxPaperSizeMetric->itemText(ui.comboBoxPaperSizeMetric->currentIndex()) == "mm") {
+			auto outputPaperSize = PaperSizeProcessor::getSizeMM(textPaperSize.toStdString());
+			ui.lineEditPaperSizeWidth->setText(QString::number(outputPaperSize.width));
+			ui.lineEditPaperSizeHeight->setText(QString::number(outputPaperSize.height));
+		} else if(ui.comboBoxPaperSizeMetric->itemText(ui.comboBoxPaperSizeMetric->currentIndex()) == "inch") {
+			auto outputPaperSize = PaperSizeProcessor::getSizeInch(textPaperSize.toStdString());
+			QLocale locale;
```

Comment:
This is a general question. currenty I've always used QString::number, which is not locale aware. If we want to use locale-aware number representation, then we should do so everywhere.

Reply:
I think we can support locale-dependent numbers here. And wiil be great, if we will extend this support to other parts of gIR